lib/skate/detours/notification_scheduler.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Skate.Detours.NotificationScheduler do
+  @moduledoc """
+  Interface for managing detour expiration notifiactions.
+  """
+
+  alias Skate.Detours.Db.{Detour, DetourExpirationNotification}
+  require Logger
+
+  def detour_activated(
+        %Detour{status: :active, estimated_duration: estimated_duration} = detour,
+        expires_at
+      ) do
+    %{detour: detour, estimated_duration: estimated_duration, expires_at: expires_at}
+    |> DetourExpirationNotification.create_changeset()
+    |> Skate.Repo.insert()
+  end
+
+  def detour_activated(_, _), do: :error
+
+  def detour_deactivated(%Detour{status: :past} = detour) do
+    DetourExpirationNotification
+    |> Skate.Repo.get_by!(detour_id: detour.id)
+    |> Skate.Repo.delete()
+  end
+
+  def detour_deactivated(_), do: :error
+
+  def detour_duration_changed(
+        %Detour{status: :active, estimated_duration: estimated_duration} = detour,
+        expires_at
+      ) do
+    DetourExpirationNotification
+    |> Skate.Repo.get_by(detour_id: detour.id)
+    |> DetourExpirationNotification.update_changeset(%{
+      detour: detour,
+      estimated_duration: estimated_duration,
+      expires_at: expires_at
+    })
+    |> Skate.Repo.insert_or_update()
+  end
```

Comment:
suggestion(non-blocking): https://hexdocs.pm/elixir/code-anti-patterns.html#complex-extractions-in-clauses

```suggestion
  def detour_activated(
        %Detour{status: :active} = detour,
        expires_at
      ) do
    %{estimated_duration: estimated_duration} = detour

    %{detour: detour, estimated_duration: estimated_duration, expires_at: expires_at}
    |> DetourExpirationNotification.create_changeset()
    |> Skate.Repo.insert()
  end

  def detour_activated(_, _), do: :error

  def detour_deactivated(%Detour{status: :past} = detour) do
    DetourExpirationNotification
    |> Skate.Repo.get_by!(detour_id: detour.id)
    |> Skate.Repo.delete()
  end

  def detour_deactivated(_), do: :error

  def detour_duration_changed(
        %Detour{status: :active} = detour,
        expires_at
      ) do
    %{estimated_duration: estimated_duration} = detour

    DetourExpirationNotification
    |> Skate.Repo.get_by(detour_id: detour.id)
    |> DetourExpirationNotification.update_changeset(%{
      detour: detour,
      estimated_duration: estimated_duration,
      expires_at: expires_at
    })
    |> Skate.Repo.insert_or_update()
  end
```

Reply:
a little more clear
```diff
Commit ID: b8c5fc1b380cdd3f936098e3bfc2323348c5de06
Change ID: vovmyrmwwrzxppypmwzpruztolkvtksm
Author   : Kayla Firestack <kfirestack@mbta.com> (2025-05-28 10:00:05)
Committer: Kayla Firestack <kfirestack@mbta.com> (2025-05-28 10:02:15)

    (no description set)

diff --git a/lib/skate/detours/notification_scheduler.ex b/lib/skate/detours/notification_scheduler.ex
index 44755daef2..d19000eebc 100644
--- a/lib/skate/detours/notification_scheduler.ex
+++ b/lib/skate/detours/notification_scheduler.ex
@@ -7,9 +7,11 @@
   require Logger
 
   def detour_activated(
-        %Detour{status: :active, estimated_duration: estimated_duration} = detour,
+        %Detour{status: :active} = detour,
         expires_at
       ) do
+    %{estimated_duration: estimated_duration} = detour
+
     %{detour: detour, estimated_duration: estimated_duration, expires_at: expires_at}
     |> DetourExpirationNotification.create_changeset()
     |> Skate.Repo.insert()
@@ -26,9 +28,11 @@
   def detour_deactivated(_), do: :error
 
   def detour_duration_changed(
-        %Detour{status: :active, estimated_duration: estimated_duration} = detour,
+        %Detour{status: :active} = detour,
         expires_at
       ) do
+    %{estimated_duration: estimated_duration} = detour
+
     DetourExpirationNotification
     |> Skate.Repo.get_by(detour_id: detour.id)
     |> DetourExpirationNotification.update_changeset(%{
```